source/blog/2018-03-02-containers-w-skopeo-ostree-oci.md

```diff
@@ -0,0 +1,310 @@
+and run them the same way as the [Atomic tool](https://github.com/projectatomic/atomic) does.
+
+If you inspected the `flannel` container image (either using `docker inspect` or remotely with `skopeo inspect`)
+you would see that it has a label called `atomic.type` indicating it is a system container.
+
+```
+$ skopeo inspect docker://registry.fedoraproject.org/f27/flannel
+{
+    "Name": "registry.fedoraproject.org/f27/flannel",
+    "Labels": {
+        "atomic.type": "system",
+// ...
+```
+
+Either that or by passing `--system` after `atomic install`,
+those are special containers that are executed without Docker daemon,
+those containers have [a special directory structure](http://www.projectatomic.io/blog/2016/09/intro-to-system-containers/)
+like their `systemd` service template as you can see in the source of [Fedora's flannel container source](https://src.fedoraproject.org/container/flannel/blob/master/f/Dockerfile#_23).
+
+The steps in this article are inspired by [how atomic tool work under the hood](https://github.com/projectatomic/atomic/blob/v1.22/Atomic/syscontainers.py).
+
+You can follow those steps on atomic host or in your regular OS (I've tested them on regular Fedora Workstation),
+and you don't need to be root.
+
+## OSTree - a space-efficient way to store images locally
+
+OSTree is the same technology used by Atomic host to store its own host OS images.
+It's a content-addressable object storage to store files,
+which means a file is stored once even if it's in multiple images,
+this is even more efficient than layer-based Docker's storage backends, because it's not on layer level, but on file level.
+
+Let's start by creating a directory and initializing it to contain bare OSTree repo,
+but because we are running as non-root we need to pass `--mode=bare-user` instead of `--mode=bare`
+
+```
+$ mkdir ostree
+$ cd ostree
+$ ostree init --mode=bare-user --repo=$PWD
+```
+
+## Skopeo - for dealing with container Images and Image registries
+
+Skopeo can inspect remote container images from various registries and formats,
+pull them, and store them in many kinds of ways. 
+We are going to demonstrate how to pull small images and run them,
+so for this purpose let's choose some small few megabytes images like `docker://redis:alpine`.
+
+```
+$ skopeo copy docker://redis:alpine ostree:redis@$PWD
+$ skopeo copy docker://nginx:alpine ostree:nginx@$PWD
+$ skopeo copy docker://busybox:alpine ostree:busybox@$PWD
+```
+
+You can list images in OSTree using:
+
+```
+$ ostree refs
+```
+
+The interesting part of the output looks like:
+
+```
+ociimage/redis_3Alatest
+ociimage/nginx_3Alatest
+ociimage/busybox_3Alatest
+```
+
+The Atomic command like tool is written in python, and it uses `libostree` via `gobject-introspection`, it looks like [this](https://github.com/projectatomic/atomic/blob/v1.22/Atomic/syscontainers.py#L26).
+
+```
+import gi
+gi.require_version('OSTree', '1.0')
+from gi.repository import OSTree
+```
```

Comment:
Lines 118-124 seem out of place in the overall flow of the article.  Is there a reason to include this snippet of code while discussing using the `ostree` CLI?

Reply:
it says that atomic which is written python uses ostree via `gi` not using something like `subprocess`